wasm/store.go

```diff
@@ -1765,57 +1767,39 @@ func (s *Store) AddHostFunction(moduleName, funcName string, fn reflect.Value) e
 	if err := s.engine.Compile(f); err != nil {
 		return fmt.Errorf("failed to compile %s: %v", f.Name, err)
 	}
-	m.Exports[funcName] = &ExportInstance{Kind: ExportKindFunc, Function: f}
 	s.addFunctionInstance(f)
-	return nil
+
+	return m.addExport(funcName, &ExportInstance{Kind: ExportKindFunc, Function: f})
```

Comment:
This is done at the bottom to save code duplication. Ack this instantiates a conflicting function for no reason, but I think saving code and duplication is worth it!

If there is something we need to undo on err I don't mind adding that.

Reply:
actually I'll do that now!